zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -712,24 +888,46 @@ mod test {
         eth_types::Transaction::from(mock_transaction)
     }
 
-    #[test]
-    fn begin_tx_gadget_simple() {
+    fn begin_tx_gadget_simple(enable_skipping_invalid_tx: bool) {
         // Transfer 1 ether to account with empty code, successfully
-        test_ok(mock_tx(eth(1), gwei(2), vec![]), None);
+        test_ok(
+            mock_tx(eth(1), gwei(2), vec![]),
+            None,
+            enable_skipping_invalid_tx,
+        );
 
         // Transfer 1 ether, successfully
-        test_ok(mock_tx(eth(1), gwei(2), vec![]), Some(code_with_return()));
+        test_ok(
+            mock_tx(eth(1), gwei(2), vec![]),
+            Some(code_with_return()),
+            enable_skipping_invalid_tx,
+        );
 
         // Transfer 1 ether, tx reverts
-        test_ok(mock_tx(eth(1), gwei(2), vec![]), Some(code_with_revert()));
+        test_ok(
+            mock_tx(eth(1), gwei(2), vec![]),
+            Some(code_with_revert()),
+            enable_skipping_invalid_tx,
+        );
 
         // Transfer nothing with some calldata
         test_ok(
             mock_tx(eth(0), gwei(2), vec![1, 2, 3, 4, 0, 0, 0, 0]),
             Some(code_with_return()),
+            enable_skipping_invalid_tx,
         );
     }
 
+    #[test]
+    fn begin_tx_gadget_simple_enable_skipping_invalid_tx() {
+        begin_tx_gadget_simple(true);
+    }
+
+    #[test]
+    fn begin_tx_gadget_simple_disable_skipping_invalid_tx() {
+        begin_tx_gadget_simple(false);
+    }
+
```

Comment:
I think above two test can grouped into single test with true/false because both of them doen't include any invalid transaction

Also below some test seems can be group as well

Reply:
Grouped those tests.